src/main/java/frc/robot/vision/VisionFilters.java

```diff
@@ -0,0 +1,56 @@
+package frc.robot.vision;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import frc.robot.constants.Field;
+import frc.robot.vision.data.AprilTagVisionData;
+import frc.robot.vision.data.VisionData;
+import frc.utils.Filter;
+
+public class VisionFilters {
+
+	public static Filter<VisionData> isPitchInTolerance(Rotation2d pitchTolerance) {
+		return new Filter<>(
+			visionData -> Math.abs(visionData.getEstimatedPose().getRotation().getY()) <= pitchTolerance.getRadians()
+				|| Math.abs(visionData.getEstimatedPose().getRotation().getY()) >= Math.toRadians(360) - pitchTolerance.getRadians()
+		);
+	}
+
+	public static Filter<VisionData> isRollInTolerance(Rotation2d rollTolerance) {
+		return new Filter<>(
+			visionData -> Math.abs(visionData.getEstimatedPose().getRotation().getX()) <= rollTolerance.getRadians()
+				|| Math.abs(visionData.getEstimatedPose().getRotation().getX()) >= Math.toRadians(360) - rollTolerance.getRadians()
```

Comment:
not Math.toRadians(360), I am quite sure there is a constant somewhere of a full circle or 360 or 2π